test/e2e/json-rpc/wallet_addEthereumChain.spec.ts

```diff
@@ -434,4 +434,69 @@ describe('Add Ethereum Chain', function () {
       );
     });
   });
+  describe('the dapp is already permitted to use the chain being added, and the dapp is on the same chain as the chain being added, but the rpcEndpoint being proposed does not match any existing rpcEndpoints for the chain', () => {
+    it('prompts to add the rpc endpoint to the chain networkConfiguration and set it as the default', async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withNetworkControllerDoubleGanache()
+            .withPermissionControllerConnectedToTestDappWithChains(['0x539'])
+            .build(),
+          ganacheOptions: {
+            ...defaultGanacheOptions,
+            concurrent: [{ port: 8546, chainId: '0x539' }],
```

Comment:
don't think you need double ganache nor the second ganache instance since you can target localhost and 127.0.0.1

Reply:
https://github.com/MetaMask/metamask-extension/pull/29885/commits/e087e519199e717e63778be973a69fac37c513d5